File: back/app/history/domain/DTO/historyDTO.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel
from uuid import UUID

from app.model.domain.DTO.modelDTO import ModelDTO


class HistoryDTO(BaseModel):
    id: UUID
    room_id: Optional[UUID] = None
    image_id: Optional[UUID] = None
    room_name: Optional[str] = None
    scanned_at: datetime
    model: Optional[ModelDTO] = None

File: front/api/DTO/history.dto.ts
export interface ModelLightDTO {
	id: string | null;
	name: string | null;
}

export default interface HistoryDTO {
	id: string;
	image_id?: string | null;
	room_name?: string | null;
	scanned_at: string;
	model?: ModelLightDTO | null;
}

